### PR TITLE
GEODE-2306: Update native client BUILDING.md to reflect changes for Geode

### PR DESCRIPTION
**Pull request by PivotalSarge**
- Replaced GEMFIRE_HOME with GEODE_ROOT.
- Corrected build instructions.
- Minor copy-editing.

### CODE REVIEW

File: src/BUILDING.md [hunk 1]
```diff
@@ -1,8 +1,13 @@
 # Building
     $ cd <clone>
-    $ mdkir build
-    $ cmake ../src
-    $ cmake --build .
+    $ mkdir build
+    $ cd build
+    $ cmake -DGEODE_ROOt=/path/to/geode \
```
Comment (metatype) [inline src/BUILDING.md]: Look like the 't' in `ROOt` needs to be capitalized?

File: src/BUILDING.md [hunk 2]
```diff
@@ -17,7 +22,7 @@
 * C++11 compiler *(see platform specific requirements)*
 
 ### Optional Tools
-* [Doxygen 8.11](http://www.stack.nl/~dimitri/doxygen/download.html) *(If building source documentation)*
+* [Doxygen 8.11](http://www.stack.nl/~dimitri/doxygen/download.html) *(for building source documentation)*
```
Comment (metatype) [inline src/BUILDING.md]: I found that doxygen was required in order to run `cmake ../src`.  If it wasn't installed cmake died.

**Comment by PivotalSarge** [comment]
Need to fix a few things.

**Comment by PivotalSarge** [comment]
Mark Doxygen as required and fix typo.

**Comment by metatype** [comment]
LGTM

**Comment by pivotal-jbarrett** [comment]
I would suggest a little more detail on the building in windows since the command provided will use NMake and suck for the person building. Include statement about -G generator options. Maybe include links to basic CMake documentation. Maybe under each platform provide examples of building on that platform. 

I would also pull out the installation directory thing to a footnote for about `make install` as it is not necessary for build or package targets.

**Comment by pivotal-jbarrett** [comment]
Let's also move the BUILDING.md down to the root directory so it is easier to find.

**Comment by upthewaterspout** [comment]
The [geode examples](https://github.com/apache/geode-examples) look for a GEODE_HOME environment variable. I think maybe some of the scripts in the java project do as well? It would be nice to be consistent and use that instead of GEODE_ROOT.

**Comment by pivotal-jbarrett** [comment]
@upthewaterspout, Its an easy change. We went for consistency with the CMake community on the explicit variable setting. It is however consistent to have the module "find" the product by environment variables if X_ROOT is not set. In our case we are looking for env:GEODE_ROOT and env:GEODE, again consistent with most CMake modules. It would be pretty easy to add env:GEODE_HOME to the list. Open a JIRA please!

**Comment by upthewaterspout** [comment]
@pivotal-jbarrett  - Here's a jira - https://issues.apache.org/jira/browse/GEODE-2317.

**Comment by PivotalSarge** [comment]
This closes #341

**Pull request closed without merging**